### PR TITLE
Move APIRule to Oathkeeper Rule CR

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Move APIRule to Oathkeeper Rule CR

**Related issue(s)**

### CODE REVIEW

File: resources/kcp/charts/kyma-environment-broker/templates/oathkeeper-rule.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+apiVersion: oathkeeper.ory.sh/v1alpha1
+kind: Rule
+metadata:
```
Comment (ebensom) [inline resources/kcp/charts/kyma-environment-broker/templates/oathkeeper-rule.yaml:3]: Missing finalizer:
```
  finalizers:
  - finalizer.oathkeeper.ory.sh
```

**Review by piotrmiskiewicz — APPROVED** [review]
lgtm

**Review by ebensom — COMMENTED** [review]
You would also need to expose KEB throuh ory oathkeeper proxy on the kcp-gateway using a `VirtualService`. Example (not templated), excludes the new orchestration endpoints:

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: kcp-kyma-environment-broker-cp
  namespace: kcp-system
spec:
  gateways:
  - kcp-gateway.kcp-system.svc.cluster.local
  hosts:
  - kyma-env-broker.cp.stage.kyma.cloud.sap
  http:
  - corsPolicy:
      allowHeaders:
      - Authorization
      - Content-Type
      - '*'
      allowMethods:
      - GET
      - POST
      - PUT
      - DELETE
      allowOrigin:
      - '*'
    match:
    - uri:
        regex: /oauth/([^/]+/)?v2/.*
    route:
    - destination:
        host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
        port:
          number: 4455
      weight: 100
  - corsPolicy:
      allowHeaders:
      - Authorization
      - Content-Type
      - '*'
      allowMethods:
      - GET
      - POST
      - PUT
      - DELETE
      allowOrigin:
      - '*'
    match:
    - uri:
        regex: /info/runtimes
    route:
    - destination:
        host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
        port:
          number: 4455
      weight: 100
```

File: resources/kcp/charts/kyma-environment-broker/templates/oathkeeper-rule.yaml (hunk 1 above)
Comment (crabtree) [reply resources/kcp/charts/kyma-environment-broker/templates/oathkeeper-rule.yaml:3]: Isn't it automatically applied?

Comment (ebensom) [reply resources/kcp/charts/kyma-environment-broker/templates/oathkeeper-rule.yaml:3]: Hm, yea, you are right, oathkeeper-maeaster should apply these automatically if missing.

**Review by ebensom — DISMISSED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]
lgtm

**Pull request merged**